### PR TITLE
[58] ConnectionEditPart never release semantic listeners

### LINKED ISSUE #58: ConnectionEditPart never release semantic listeners
It seems that org.eclipse.gmf.runtime.diagram.ui.editparts.ConnectionEditPart never call the removeSemanticListeners() method. It is a part of a memory leak.

A pull request is coming soon for this.

### PR DESCRIPTION
**Pull request by pdeville2**
see https://github.com/eclipse-gmf-runtime/gmf-runtime/issues/58

Bug #58
Signed-off-by: Pauline DEVILLE <pauline.deville@cea.fr>

### CODE REVIEW

**Comment by pdeville2** [comment]
Sorry for the double patch, one for semantic and one for notation

File: bundles/org.eclipse.gmf.runtime.diagram.ui/src/org/eclipse/gmf/runtime/diagram/ui/editparts/ConnectionEditPart.java [hunk 1]
```diff
@@ -383,6 +385,16 @@ public void deactivate() {
         boolean wasActive = isActive();
         super.deactivate();
 		removeNotationalListeners();
+
+        EObject semanticProxy = ((View) getModel()).getElement();
+        EObject semanticElement = EMFCoreUtil.resolve(getEditingDomain(),
+            semanticProxy);
```
Comment (pcdavid) [inline bundles/org.eclipse.gmf.runtime.diagram.ui/src/org/eclipse/gmf/runtime/diagram/ui/editparts/ConnectionEditPart.java]: I'm not sure about this part. We're deactivating the edit part, it does not seem like a safe context to resolve proxies and possibly trigger (re)loading of EMF resources.

`removeSemanticListeners()` (at least the base version in GMF itself) only calls `removeListenerFilter("SemanticModel")`, and `removeListenerFilter()` seems [safe to call even on unknown/already unregistered listeners](https://github.com/eclipse-gmf-runtime/gmf-runtime/blob/master/bundles/org.eclipse.gmf.runtime.diagram.ui/src/org/eclipse/gmf/runtime/diagram/ui/editparts/ConnectionEditPart.java#L987).

Maybe we can simply call `removeSemanticListeners();` and `removeListenerFilter(SEMANTIC_PROXY);` unconditionally in this context?

Comment (pdeville2) [reply bundles/org.eclipse.gmf.runtime.diagram.ui/src/org/eclipse/gmf/runtime/diagram/ui/editparts/ConnectionEditPart.java]: Yes of course your right, sorry for this. I will change this.

**Review by pcdavid — APPROVED** [review]

**Comment by pcdavid** [comment]
Change merged via https://github.com/eclipse-gmf-runtime/gmf-runtime/pull/66 as I wanted to squash your 2 commits into a single one but I can't push the result into your own fork, and can not push directly to master either, so I needed to create a new PR...

Anyway, it's merged and will be part of 1.16.4 for 2024-06. I'll build and publish a new RC which will probably be the final version. No other changes are planned but I prefer keeping my options open in case e.g. GEF makes some late changes that require adaptations on the GMF Runtime side.

Thanks for the contribution.

**Comment by pcdavid** [comment]
It's available in https://download.eclipse.org/modeling/gmp/gmf-runtime/updates/milestones/S202405171447

**Comment by pdeville2** [comment]
Thank you @pcdavid

**Pull request closed without merging**